native/Avalonia.Native/src/OSX/main.mm

```diff
@@ -20,6 +20,20 @@ virtual HRESULT SetShowInDock(int show)  override
     {
         ShowInDock = show;
         SetActivationPolicy();
+        
+        
+        /* carbon voodoo to get icon and menu without bundle */
+        ProcessSerialNumber psn = { 0, kCurrentProcess };
+        TransformProcessType(&psn, kProcessTransformToForegroundApplication);
+        SetFrontProcess(&psn);
+        
+        id menubar = [NSMenu new];
+        [menubar setTitle:@"Test"];
+        [NSApp setMainMenu:menubar];
+        id appName = [[NSProcessInfo processInfo] processName];
```

Comment:
We probably need to get it from the user code and only fall back to processName when it's not provided